library/azure_rm_devtestlabslab.py

```diff
@@ -0,0 +1,281 @@
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_devtestlabslab
+version_added: "2.8"
+short_description: Manage Azure DevTest Lab instance.
+description:
+    - Create, update and delete instance of Azure DevTest Lab.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group.
+        required: True
+    name:
+        description:
+            - The name of the lab.
+        required: True
+    location:
+        description:
+            - The location of the resource.
+    lab_storage_type:
+        description:
+            - Type of storage used by the lab. It can be either C(premium) or C(standard). Default is C(premium).
+        choices:
+            - 'standard'
+            - 'premium'
+        default: premium
+    premium_data_disks:
+        description:
+            - "Allow creation of C(premium) data disks."
+        type: bool
+    state:
+      description:
+        - Assert the state of the DevTest Lab.
+        - Use 'present' to create or update an DevTest Lab and 'absent' to delete it.
+      default: present
+      choices:
+        - absent
+        - present
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Create (or update) DevTest Lab
+    azure_rm_devtestlabslab:
+      resource_group: testrg
+      name: mylab
+      lab_storage_type: standard
+'''
+
+RETURN = '''
+id:
+    description:
+        - The identifier of the resource.
+    returned: always
+    type: str
+    sample: id
+'''
+
+import time
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+from ansible.module_utils.common.dict_transformations import _snake_to_camel
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrest.polling import LROPoller
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.mgmt.devtestlabs import DevTestLabsClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMDevTestLab(AzureRMModuleBase):
+    """Configuration class for an Azure RM DevTest Lab resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str'
+            ),
+            lab_storage_type=dict(
+                type='str',
+                choices=['standard',
+                         'premium'],
+                default='premium'
```

Comment:
default value is dangerous as we disucssed. such as at first run, user set storage_type as `standard`, at second run, user didn't specify this argument, because they don't want to change it, then comparing default  `premium` not equal to `standard`, change happened.